compiler/yaml/src/main/core/com/kitakkun/backintime/compiler/yaml/CallableSignature.kt

```diff
@@ -0,0 +1,120 @@
+
+    override fun deserialize(decoder: Decoder): CallableSignature {
+        val value = decoder.decodeString()
+
+        return if (value.startsWith("<get-") && value.endsWith(">")) {
+            val propertyName = value.removePrefix("<get-").removeSuffix(">")
+            CallableSignature.PropertyAccessor.Getter(propertyName)
+        } else if (value.startsWith("<set-") && value.endsWith(">")) {
+            val propertyName = value.removePrefix("<set-").removeSuffix(">")
+            CallableSignature.PropertyAccessor.Setter(propertyName)
+        } else if (value == "<this>") {
+            CallableSignature.This
+        } else {
+            // com/example/ReceiverClass com/example/hoge(*, kotlin/Int)
+            // -> com/example/ReceiverClass com/example/hoge
+            val signature = value.takeWhile { it != '(' }
+            val receiverAndFunction = signature.split(" ")
+
+            val receiverPart: String
+            val functionPart: String
+            when (receiverAndFunction.size) {
+                1 -> {
+                    receiverPart = ""
+                    functionPart = receiverAndFunction.single()
+                }
+
+                2 -> {
+                    receiverPart = receiverAndFunction[0]
+                    functionPart = receiverAndFunction[1]
+                }
+
+                else -> error("Error!")
+            }
+
+            val functionName = functionPart.takeLastWhile { it != '.' && it != '/' }
+            val functionPackage = functionPart.removeSuffix(functionName)
+                .replace("/", ".")
+                .removeSuffix(".")
+
+            val parameters = if (!value.contains("(")) {
+                ParametersSignature.Any
+            } else {
+                ParametersSignature.Specified(
+                    value.dropWhile { it != '(' }.dropLastWhile { it != ')' }
+                        .split(",")
+                        .map { it.trim() }
+                        .map {
+                            if (it == "*") TypeSignature.Any
+                            else if (it.toIntOrNull() != null) TypeSignature.Generic(it.toInt())
+                            else TypeSignature.Class(it)
+                        }
+                )
+            }
+
+            if (receiverPart.isEmpty() && functionPackage.isEmpty()) {
+                CallableSignature.NamedFunction.Member(
+                    name = functionName,
+                    valueParameters = parameters,
+                )
+            } else {
+                CallableSignature.NamedFunction.TopLevel(
+                    receiverClassId = receiverPart,
+                    packageFqName = functionPackage,
+                    name = functionName,
+                    valueParameters = parameters,
+                )
+            }
+        }
+    }
+
+    override fun serialize(encoder: Encoder, value: CallableSignature) {
+    }
```

Comment:
[nits] Better to implement `serialize` function as well. Though it is not necessarily required, the implementation like this might confuse us in the future.

Reply:
done [Make yaml serializable](https://github.com/kitakkun/back-in-time-plugin/pull/138/commits/9a412694ed8f9946f20a52c57bb6787e1edc4288)